lualib/resty/events/worker.lua

```diff
@@ -243,21 +244,23 @@ function _M:communicate(premature)
             sleep(0)
 
             ::continue::
-        end -- while not exiting
-    end)  -- events_thread
+        end -- while not exiting or restarting
+    end) -- events_thread
 
     local ok, err, perr = wait(write_thread, read_thread, events_thread)
```

Comment:
Should we just wait do like this:

```suggestion
    local ok, err, perr = wait(write_thread, read_thread)
```